bootstrap.py

```diff
@@ -32,7 +32,7 @@ def install_with_pip(packages):
 def detect(setting):
     """ Detects a setting in tags, falls back to environment variables """
     import os
-    if setting in resource_tags():
+    if resource_tags() is not None and setting in resource_tags():
```

Comment:
`detect()` should not be aware of how `resource_tags()` is implemented. This is an inconsistent interface, and should be resolved in the callee, not worked around in the caller.